messaging/broker.go

```diff
@@ -332,67 +348,249 @@ func (b *Broker) applyUnsubscribe(m *Message) {
 // This is implemented as a separate type because it is not meant to be exported.
 type brokerFSM Broker
 
-// Apply executes a raft log entry against the broker.
-func (fsm *brokerFSM) Apply(e *raft.LogEntry) error {
+// MustApply executes a raft log entry against the broker.
+// Non-repeatable errors such as system or disk errors must panic.
+func (fsm *brokerFSM) MustApply(e *raft.LogEntry) {
 	b := (*Broker)(fsm)
 
-	// Ignore internal raft entries.
-	if e.Type != raft.LogEntryCommand {
-		// TODO: Save index.
-		return nil
-	}
+	// Save highest applied index.
+	// TODO: Persist to disk for raft commands.
+	b.index = e.Index
```

Comment:
I might be missing something here, but why do we save this index before the "apply" code below? As I understand it "applied" means log entries that have changed the state machine. But doesn't that only happen below?

(The comments on the Broker struct state that `index` is the "highest applied index.)